azext_iot/iothub/providers/message_endpoint.py

```diff
@@ -224,6 +218,226 @@ def create(
         except HttpResponseError as e:
             handle_service_exception(e)
 
+    def update(
+        self,
+        endpoint_name: str,
+        endpoint_type: str,
+        endpoint_account_name: Optional[str] = None,
+        endpoint_resource_group: Optional[str] = None,
+        endpoint_subscription_id: Optional[str] = None,
+        endpoint_policy_name: Optional[str] = None,
+        connection_string: Optional[str] = None,
+        container_name: Optional[str] = None,
+        encoding: Optional[str] = None,
+        batch_frequency: Optional[int] = None,
+        chunk_size_window: Optional[int] = None,
+        file_name_format: Optional[str] = None,
+        endpoint_uri: Optional[str] = None,
+        entity_path: Optional[str] = None,
+        database_name: Optional[str] = None,
+        primary_key: Optional[str] = None,
+        secondary_key: Optional[str] = None,
+        partition_key_name: Optional[str] = None,
+        partition_key_template: Optional[str] = None,
+        identity: Optional[str] = None
+    ):
+        # if nothing is provided -> should we block?
```

Comment:
Question. Currently other update commands do not block if no arguments are provided (as far as I know)

Reply:
block as in do nothing/do not call the api

so in the current scenario, the call is still made (like how other update calls function)

Reply:
Gotcha, yeah looks like in `az iot hub update` we still do call the GET -> PUT even if no parameters have changed.

I don't think this is an issue, especially if it's similar to how other commands currently function.